skiko/src/jvmMain/kotlin/org/jetbrains/skiko/PlatformOperations.kt

```diff
@@ -20,6 +22,16 @@ internal val platformOperations: PlatformOperations by lazy {
                 }
 
                 override fun setFullscreen(component: Component, value: Boolean) {
+                    // MacOS specific: if fullscreen is set before the window is visible,
+                    // we will add a listener to set fullscreen after the window is visible
+                    val window = SwingUtilities.getRoot(component) as Window
+                    if (value && !window.isVisible) {
+                        window.addComponentListener(object : ComponentAdapter() {
```

Comment:
we add a listener every time when we call setFullscreen and window isn't visible:
```
val layer = SkiaLayer()
layer.isFullscreen = true
layer.isFullscreen = false
layer.isFullscreen = true
layer.isVisible = true
layer.isVisible = false
layer.isFullscreen = false
layer.isFullscreen = true
```

can we call `osxSetFullscreenNative` in `SkiaLayer.addNotify`?

Reply:
Also, this will return false:
```
val layer = SkiaLayer()
layer.isVisible = true
layer.isFullscreen = true
println(layer.isFullscreen)
```

It is an issue of Skiko, but I am not sure, if it is issue for Compose. Theoretically it can be, as we check layer.isFullscreen in resize listener, which can be triggered after we set layer.isFullscreen = true.

To fix that, we need to introduce intermediate state, as you done [here](https://github.com/JetBrains/skiko/commit/a71eca0fdd03caf248a99e4ca2efb537ead76736#diff-1782b97ce06bbcf5e55689e5c50647096808e91aef66c3a60fef15beb477441eR47), but with careful handling of concurrency.

Reply:
I would implement something like this:
```
class SkiaLayer {
        var isFullscreen: Boolean = false
           set(value) {
              field = value
              platformOperation.setFullscreen(value)
           }

        private var window: Window? = null
        private var windowListener = object : WindowAdapter {
            override fun windowResized(e: WindowEvent) {
                isFullscreen = platformOperations.isFullscreen()
            }
        }

        override fun addNotify() {
            super.addNotify()
            ...
            window = SwingUtilities.getWindowAncestor(this)
            window?.addWindowListener(windowListener)
            platformOperation.setFullscreen(isFullscreen)
        }

        override fun removeNotify() {
            window?.removeWindowListener(windowListener)
            window = null
            ...
            super.removeNotify()
        }
}
```